keps/sig-scheduling/3280-guarantee-pdb-when-preemption-happens/README.md

```diff
@@ -0,0 +1,981 @@
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `DisruptionPolicyInPriorityClass`
+  - Components depending on the feature gate: `kube-scheduler`, `kube-apiserver`
+
+###### Does enabling the feature change any default behavior?
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+No.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+
+The feature can be disabled in Alpha and Beta versions by restarting 
+kube-apiserver and kube-scheduler with feature-gate off.
+One caveat is that PriorityClasses that used the feature will continue to have the 
+field `DisruptionPolicy` set in `PriorityClass` even after disabling 
+the feature gate, however kube-scheduler will not take the field into account.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+1. The newly created PriorityClass will contain the field `DisruptionPolicy`.
+2. The scheduler will check the value in the field `DisruptionPolicy` in `PriorityClass` 
+   if preemption occurs during scheduling
+
+###### Are there any tests for feature enablement/disablement?
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+No, unit and integration tests will be added.
```

Comment:
FWIW - the only enablement/disablement test that I expect here is an api test described in the comment above.

[Do you have any other enablement/disablement tests on your mind (note that we're talking about tests that change the value of feature-gate in the middle of the test here) that would be useful here?]